WordPress/gutenberg
# Try: Sync block context with store

## What?
The idea behind this pull request is to sync the block context in the store.

## Why?
To be able to access it without using a hook for uses cases like [bindings modifying the block attributes](https://github.com/WordPress/gutenberg/pull/61817).

## How?
Creating a new action `updateBlockContext` to modify the block context in the store and a selector `getBlockContext` to retrieve it.

Apart from that, I'm syncing the context in the edit component using `useContext`.

## Testing Instructions
* In any block, use the `getBlockContext` selector and check that the block context is present.
